### ISSUE TITLE
Long item unit names can clash with added quantity in shop

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#11967]
## Description

Text can become unreadable when two labels are overlapping. See Screenshot.

## Expected Behavior

We could limit the length of the item description displayed or introduce line breaks.

## Actual Behaviour

The text overlays and becomes unreadable.

## Steps to Reproduce

1. Set up a product with items as unit.
2. Enter a very long unit name like in the screenshot.
3. Visit the shop with a small screen, like mobile.

## Animated Gif/Screenshot

![IMG_3644](https://github.com/openfoodfoundation/openfoodnetwork/assets/3524483/2fc28ab0-7cc4-4d96-8b78-89923f777edd)

## Workaround

Shorten the unit name.

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

**Comment by Isaac-alencar** [comment]
Hey, can I take this one?

**Comment by sigmundpetersen** [comment]
Sure @Isaac-alencar , thanks!
I'll assign you

**Comment by Isaac-alencar** [comment]
I was trying to use CSS `text-overflow` to hide the overlay text, but still kinda illegible. What do you think to re-order the elements displayed if the item unit name is a long text? 
Screenshot:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/58452911/5298f361-acf5-4fd1-90c7-93886ab8b567)
Do you think it is a valid solution?

**Comment by drummer83** [comment]
Could you also show an example with a short item unit name, please? Will they both change in the same way? Or will there be a mix of both designs?

**Comment by Isaac-alencar** [comment]
Sure: 
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/58452911/9e2fbd3b-afa1-4c09-b858-596e027befbc)

I was thinking to apply this only for products that item unit name have a long text

**Comment by mkllnk** [comment]
Hm, having the main input element in different places depending on a mostly unimportant label is not great. It would need to be on the right, at least.

I think that text-overflow ellipses is valid, because the quantity description should not be long. The text in the example should have been changed by the shop. When a shop does put a long text there though, I don't want it to break the design.

**Comment by RachL** [comment]
Hello @Isaac-alencar I hope this message finds you well. Are you still planning to work on this issue? Let us know if you need more info!

**Comment by drummer83** [comment]
I was just wondering if the easiest solution would be to remove the 'x in cart' label. It doesn't provide any additional information, because the amount is shown between the + and - buttons anyway. I could imagine that the label is a leftover from the old implementation of the 'add' button.

@openfoodfoundation/train-drivers-product-owners Could you please comment what you think?

**Comment by RachL** [comment]
@drummer83 I think this area is also used for the message that says there is "only X left" in stock, so I'm not sure this will fix the issue...

![image](https://github.com/user-attachments/assets/46a0a73e-3e97-4e53-ad44-6491b6d42f19)

**Comment by drummer83** [comment]
Definitely no fix for that case, you're right. But it would reduce the occurrence.

**Comment by JoeLoparco** [comment]
Hello, would it be okay if I took this bug?

**Comment by mkllnk** [comment]
Yes, @JoeLoparco, go for it.

**Comment by sigmundpetersen** [comment]
Hey @JoeLoparco hope you're doing fine 👋 

Are you still interested in working on this issue?

**Comment by krisztin** [comment]
The root cause of the issue is the left to right / column layout. That and the amount of info displayed on the `<product>` listing item. For something so varied with so many possible values a top to bottom layout would be safer and more flexible.

## Possible solutions

### 1. Working with what we have (hacky and not super pretty)
Similar to Isaac's solution from a year ago, rearrange the layout but on all screen sizes to be top to bottom. This means doing away with the 3 col layout.

The below is a rough draft after playing around with the code in Firefox's inspector tool to scope out complexity/viability. Can be hacked a bit more for better alignment (price info for instance is a bit illogical right now)

![Image](https://github.com/user-attachments/assets/500c137d-8cd4-4770-a8d7-098807a4a09f)

### 2. Starting all over
Redo the whole product card to better fit a top to bottom layout. The layout itself is easy, refactoring though...😅️

**Quick, dirty prototype**

![Image](https://github.com/user-attachments/assets/74101105-4f2e-498d-b48e-e830d98fa641)

#### Things to consider

- Stats of user screen resolutions - this would help in thinking about the layout. If most users are from mobile than a mobile first approach would make sense.
- Do all of these pieces of info need to be on the listing page? Maybe could move some of it on the popup instead.

**Comment by mkllnk** [comment]
> Redo the whole product card to better fit a top to bottom layout. The layout itself is easy, refactoring though..

Technically, changing the layout is quite simple. We don't need to worry about that cost if there's a significantly better solution.

> Do all of these pieces of info need to be on the listing page?

That's a question for @openfoodfoundation/train-drivers-product-owners or instance managers. From what I know:

* Product image: yes, important.
* Product name: yes, important.
* Product description: only abbreviated but that's important, I think.
* Supplier: Yes, important. It's a key feature that we show the "network" of enterprises.
* Attributes like 'plastic-free': not sure. I could do without. That's where the filters can be really good. And if it's that important then maybe people add it to the description as well.
* Size and price: Necessary. 😉 
* Price chart: Not used much. It's a unique feature but could go into the detailed modal, I think.
* Price per unit: Required by regulations.
* Add button: Necessary.
* Scarcity indicator: optional but helpful. User study could be interesting here.

Generally, I like the look of your new design. But I'm not a big fan of tiles because going through a two-dimensional field is more complex than scrolling through a list.  With so much detail in each tile, I find that adds to my mental load. Also, minor, changing window size makes things jump around. But I would be happy to be convinced otherwise.

**Comment by RachL** [comment]
Aligned with @mkllnk 's answers. 

FYI that scarcity indicator is a feature the shop owner can choose to turn on & off. When they turn it on, it's because they want to catch the visitor's eye during scrolling.

Before commenting on the new design, I'm sorry I didn't do this before here, but I'm only realizing now, that I have a lots of questions on this topic actually 🤕 
Do we know how many cases of long unit name we have in production? Can we make a list of these examples?

I'm wondering if what the user is trying to do here could be replaced by a `100% hand-made` label for example. Or using the description. They are not using this field, it would maybe be interesting to know why.

I'm also wondering if we should allow long unit names. As Maikel underlines, this is the price per unit, it's a legal information. The unit we are after in this example is "bulb".

**Comment by mkllnk** [comment]
> I'm also wondering if we should allow long unit names.

Exactly, just to emphasize again, this issue came only up when people used the field creatively, putting much more information in than there should be. It's not a valid use case. That's why this is labeled as s5, the lowest priority of bugs.

**Comment by slothmock** [comment]
Hi all, I'd love to take on this issue. It looks like the simplest solution might be to replace any unit names longer than 10 characters with a generic label like "Unit", just for display purposes in the price per unit area.

For example, in the first screenshot, the unit name is "hand saved, planted, harvested then finally plaited bulbs of garlic", which causes the layout to break on smaller screens. Replacing it with something shorter could help avoid overlap and keep things readable regardless of platform.

Let me know if this approach makes sense, and feel free to assign the issue to me.
